source/backend/dal/Repositories/BusinessFunctionCodeRepository.cs

```diff
@@ -38,6 +42,29 @@ public IList<PimsBusinessFunctionCode> GetAllBusinessFunctionCodes()
                 .ToList();
         }
 
+        public PimsBusinessFunctionCode Add(PimsBusinessFunctionCode pimsCode)
+        {
+            pimsCode.ThrowIfNull(nameof(pimsCode));
+
+            // Check for uniqueness on save. At any given point an active code should be unique per code type.
+            var existingCodes = Context.PimsBusinessFunctionCodes.Where(
+                    c => EF.Functions.Collate(c.Code, SqlCollation.LATIN_GENERAL_CASE_INSENSITIVE) == pimsCode.Code)
+                .ToList();
+
+            var now = DateTime.UtcNow;
+            var newCodeIsActive = !pimsCode.ExpiryDate.HasValue || pimsCode.ExpiryDate?.Date > now.Date;
+
+            // Active codes have no expiry date or they expire in the future.
+            var isDuplicate = existingCodes.Any(c => !c.ExpiryDate.HasValue || c.ExpiryDate?.Date > now.Date);
+            if (newCodeIsActive && isDuplicate)
+            {
+                throw new DuplicateEntityException("Duplicate business function code found");
```

Comment:
Is there a repo test that checks for this?

Reply:
I tried to create a test but could not get it to work because the InMemory database used in tests does not support EF functions such as Collate (which we use in various repos). 

See: https://stackoverflow.com/questions/70693625/the-collate-method-is-not-supported-because-the-query-has-switched-to-client-e

Reply:
We could refactor this repo (and others that use EF functions) to be more testable but I wonder about the ROI